### PR TITLE
[DC-325] Enable UI integration tests on alpha / staging

### PR DESCRIPTION
**Pull request by slucasbroad**
Remove `targetEnvironments` and update test for Alpha environment compatibility.

### CODE REVIEW

File: integration-tests/utils/integration-utils.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ const getTableColIndex = async (page, { tableName, columnHeader }) => {
 const getTableCellByContents = async (page, { tableName, columnHeader, textContains, isDescendant = false }) => {
   const colIndex = await getTableColIndex(page, { tableName, columnHeader })
   const baseXpath = `//*[@role="table" and @aria-label="${tableName}"]//*[@role="row"]//*[@role="cell" and @aria-colindex = "${colIndex}"]`
-  const xpath = `${baseXpath}${isDescendant ? '//*' : ''}[contains(normalize-space(.),"${textContains}")]`
+  const xpath = `${baseXpath}${isDescendant ? '//*' : ''}[text() = "${textContains}"]`
```
Comment (kyuksel) [inline integration-tests/utils/integration-utils.js]: Could you explain the reasoning behind the change above?

File: integration-tests/utils/integration-utils.js [hunk 2]
```diff
@@ -97,7 +97,7 @@ const assertRowHas = async (page, { tableName, expectedColumnValues, withKey: {
 
 const clickTableCell = async (page, { tableName, columnHeader, textContains, isDescendant = false }, options) => {
   const tableCellPath = await getTableCellByContents(page, { tableName, columnHeader, textContains, isDescendant })
-  const xpath = `${tableCellPath}//*[@role="button" or @role="link" or @role="checkbox"]`
+  const xpath = `${tableCellPath}[@role="button" or @role="link" or @role="checkbox"]`
```
Comment (kyuksel) [inline integration-tests/utils/integration-utils.js:102]: Could you explain the reasoning behind the change above?

File: integration-tests/tests/link-to-new-workspace.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const { click, clickable, fillIn, findText, noSpinnersAfter, select } = require(
 const { checkBucketAccess, testWorkspaceName } = require('../utils/integration-helpers')
 const { withUserToken } = require('../utils/terra-sa-utils')
 
-
+//test fails?
```
Comment (kyuksel) [inline integration-tests/tests/link-to-new-workspace.js]:
```suggestion
```

**Review by kyuksel — COMMENTED** [review]
A few questions and comments inline. 

For PRs modifying tests and/or test utils, let's make sure to run the [test-flakes](https://github.com/DataBiosphere/terra-ui/tree/dev/integration-tests#detecting-flakiness) tool with `RUNS` set to at least `100` for each test.

File: integration-tests/utils/integration-utils.js (hunk 1 above)
Comment (slucasbroad) [reply integration-tests/utils/integration-utils.js]: Discovered in alpha test that it picked up `patient13` before `patient1` and re-testing with `[text() = "${textContains}"]` didn't create new errors.

File: integration-tests/utils/integration-utils.js (hunk 2 above)
Comment (slucasbroad) [reply integration-tests/utils/integration-utils.js:102]: has to change the test in `catalog-utils.js` to something that exist and clickable in both dev and alpha, so tested for 'Readable Catalog Snapshot 1'. and was discovered that the name was on the same path level as @role="button"

File: integration-tests/utils/integration-utils.js (hunk 1 above)
Comment (kyuksel) [reply integration-tests/utils/integration-utils.js]: Do we need to care about the ordering of content? If not, can we do the comparison in a way where order isn't important (e.g. as sets).

**Comment by kyuksel** [comment]
@slucasbroad Did you get to run [test-flakes](https://github.com/DataBiosphere/terra-ui/tree/dev/integration-tests#detecting-flakiness) against the four tests being updated in this PR? If so, could you report pass rates?

File: integration-tests/utils/integration-utils.js [hunk 4]
```diff
@@ -75,7 +75,7 @@ const getTableColIndex = async (page, { tableName, columnHeader }) => {
 const getTableCellByContents = async (page, { tableName, columnHeader, textContains, isDescendant = false }) => {
```
Comment (petesantos) [inline integration-tests/utils/integration-utils.js]: Given that this is no longer a contains, let's rename this to `text`
```suggestion
const getTableCellByContents = async (page, { tableName, columnHeader, text, isDescendant = false }) => {
```

**Comment by slucasbroad** [comment]
Flack test pass rate
request-access
Snapshots:   0 total
Running tests: 100% 

link-to-new-workspace
Snapshots:   0 total
Running tests: 100% 

preview-dataset
Snapshots:   0 total
Time:        307.14 s

run-catalog-workflow
Snapshots:   0 total
Running tests: 100% 

Note: line 35 in `jest-reporter.js` is bugged and was commented out. 
Error gave: 
    `Test suite failed to run`

    TypeError: Cannot read properties of undefined (reading 'forEach')

**Comment by slucasbroad** [comment]
Flack test pass rate
request-access
Snapshots:   0 total
Running tests: 100% 

link-to-new-workspace
Snapshots:   0 total
Running tests: 100% 

preview-dataset
Snapshots:   0 total
Time:        307.14 s

run-catalog-workflow
Snapshots:   0 total
Running tests: 100% 

Note: line 35 in `jest-reporter.js` is bugged and was commented out. 
Error gave: 
`Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'forEach')`

**Comment by kyuksel** [comment]
> Note: line 35 in `jest-reporter.js` is bugged and was commented out. Error gave: `Test suite failed to run
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'forEach')`
> ```

Changing line 35 as 

```javascript
_.forEach(({ message }) => writableStream.write(`${message}\n`), testResult?.console)
```

may prevent the error.

**Review by kyuksel — APPROVED** [review]

**Comment by kyuksel** [comment]
@slucasbroad Does [run-catalog-filter](https://github.com/DataBiosphere/terra-ui/blob/97a79d9ca2a27b66801954f4e26ede1a4d1b2b66/integration-tests/tests/run-catalog-filter.js#L59) need to be enabled as well?

**Comment by slucasbroad** [comment]
> run-catalog-filter

**Comment by petesantos** [comment]
> > Note: line 35 in `jest-reporter.js` is bugged and was commented out. Error gave: `Test suite failed to run
> > ```
> > TypeError: Cannot read properties of undefined (reading 'forEach')`
> > ```
> 
> Changing line 35 as
> 
> ```js
> _.forEach(({ message }) => writableStream.write(`${message}\n`), testResult?.console)
> ```
> 
> may prevent the error.

I filed a ticket for this: https://broadworkbench.atlassian.net/browse/QA-1879
There was a change to stop using `_.forEach` recently, I would rather have QA look at this than thrash on the code.

**Comment by petesantos** [comment]
> @slucasbroad Does [run-catalog-filter](https://github.com/DataBiosphere/terra-ui/blob/97a79d9ca2a27b66801954f4e26ede1a4d1b2b66/integration-tests/tests/run-catalog-filter.js#L59) need to be enabled as well?

This test will need some more changes: https://broadworkbench.atlassian.net/browse/QA-1879

**Review by petesantos — APPROVED** [review]

File: integration-tests/jest-reporter.js [hunk 5]
```diff
@@ -32,8 +32,7 @@ module.exports = class JestReporter {
     const writableStream = createWriteStream(logFileName)
     writableStream.on('error', ({ message }) => console.error(`Error occurred while writing Console logs to ${logFileName}.\n${message}`))
 
-    testResult.console.forEach(({ message }) => writableStream.write(`${message}\n`)
```
Comment (aweng98) [inline integration-tests/jest-reporter.js:35]: @slucasbroad Hi, please revert this change. `_.forEach` fails to get all console logs. Issue was fixed in https://github.com/DataBiosphere/terra-ui/pull/3024

**Pull request merged**